nemo/collections/tts/models/fastpitch.py

```diff
@@ -131,6 +131,9 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self.duration_loss_fn = DurationLoss(loss_scale=dur_loss_scale)
         self.energy_loss_fn = EnergyLoss(loss_scale=energy_loss_scale)
 
+        if "pitch_stats_path" in cfg:
```

Comment:
It is not clear to me why do we need to register pitch_stats_path as an artifact of the model? Pitch stats is a feature derived from the data and not the model. Also, if we do register it as an artifact, won't it throw errors when we try to load checkpoints which were trained using `pitch_mean`, `pitch_std` ... variables?

Reply:
I was trying to copy the convention of the pitch_mean/pitch_std in which in the .yaml it configures it at the top level, then saves it in the model, and then passes it from model to data loader. It then saves some value in the module, but I cannot tell if that actually does anything. https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/modules/fastpitch.py#L189-L190

Register artifact copies the artifact into the .nemo file and then replaces the path on a user's local machine with the relative path in the checkpoint. It will have no impact if that config is not provided. If you do not register it, then loading the checkpoint breaks whenever you load it on a different machine (or move the path the file as at). So I think registering it somewhere is necessary.

I think this does imply that you cannot change the convention from the original training. So if you train with the hardcoded mean/std, you cannot fine tune with the dictionary based config. Or vice versa.

Reply:
After testing it, I don't think register artifact matters for training or fine-tuning. It is only used when loading the model for inference, which does not need the data loader configs.

So, removing the call to register_artifact.

Reply:
sure.